Sources/NIO/ChannelOption.swift

```diff
@@ -327,3 +327,39 @@ extension ChannelOptions {
         }
     }
 }
+
+// Provides convenience methods for `ChannelOption.socket`
+extension ChannelOptions {
+    
+    public struct Socket {
+        
+        /// Turn on debugging info recording
+        public static let enableDebugging = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_DEBUG)
+        
+        /// Socket has had listen()
+        public static let acceptConnections = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_ACCEPTCONN)
+        
+        /// Allow local address reusue
+        public static let allowLocalAddressReuse = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_REUSEADDR)
+        
+        /// Keep connections alive
+        public static let keepAlive = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_KEEPALIVE)
+        
+        /// Just use interface addresses
+        public static let dontRoute = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_DONTROUTE)
+        
+        /// Permit sending of broadcast messages
+        public static let enableBroadcastMessages = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_BROADCAST)
+        
+        /// Bypass hardware when possible
+        public static let useLoopback = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_USELOOPBACK)
+        
+        /// Linger on close if data present
+        public static let linger = ChannelOptions.Types.SocketOption(level: SOL_SOCKET, name: SO_LINGER)
```

Comment:
Let's not bother with `SO_LINGER`, as `SocketOptionProvider` is a better API for setting it.